### PR TITLE
feat(http): add sample http service calls

### LINKED ISSUE #839: More examples?
Is it within the scope of this project to add more examples? For instance how to mock a http request for unit tests, how to use mock a class for testing etc.

Or do you think this is better handled else where?

Obviously i am happy to contribute towards these.

### PR DESCRIPTION
**Pull request by mgechev**
Fix #839

### CODE REVIEW

**Comment by ludohenin** [comment]
Awesome! Will run and merge later today.

File: src/client/app/app.component.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+import { HTTP_PROVIDERS} from '@angular/http';
 
 import { AboutComponent } from './+about/index';
 import { HomeComponent } from './+home/index';
 import { NameListService, NavbarComponent, ToolbarComponent } from './shared/index';
 
 @Component({
   selector: 'sd-app',
-  viewProviders: [NameListService],
+  viewProviders: [NameListService, HTTP_PROVIDERS],
```
Comment (ludohenin) [inline src/client/app/app.component.ts]: Just for the sake of consistency (and probably best practice), I'd place `HTTP_PROVIDERS` in `main.ts` together with the router ones.

File: src/client/app/shared/name-list/name-list.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,46 @@
+import { ReflectiveInjector, provide } from '@angular/core';
```
Comment (ludohenin) [inline src/client/app/shared/name-list/name-list.service.spec.ts]: `ReflectiveInjector, is it a replacement of `Injector` or something else ?

File: src/client/app/+home/home.component.spec.ts [hunk 3]
```diff
@@ -8,6 +8,14 @@ import {
 } from '@angular/core/testing';
 import { getDOM } from '@angular/platform-browser/src/dom/dom_adapter';
 
+import { MockBackend } from '@angular/http/testing';
```
Comment (TheDonDope) [inline src/client/app/+home/home.component.spec.ts]: https://angular.io/styleguide#!#coding-conventions
> Import Line Spacing
> Style 03-06
> Do list import lines alphabetized by the module.

File: src/client/app/shared/name-list/name-list.service.spec.ts [hunk 4]
```diff
@@ -1,16 +1,46 @@
...
+import { MockBackend } from '@angular/http/testing';
+import { HTTP_PROVIDERS, BaseRequestOptions, Http, Response, ResponseOptions, ConnectionBackend } from '@angular/http';
```
Comment (TheDonDope) [inline src/client/app/shared/name-list/name-list.service.spec.ts]: https://angular.io/styleguide#!#coding-conventions
> Import Line Spacing
> Style 03-06
> Do list destructured imported assets alphabetically.

File: src/client/app/shared/name-list/name-list.service.spec.ts [hunk 5]
```diff
@@ -1,16 +1,46 @@
...
 import { NameListService } from './name-list.service';
 
+import { Observable } from 'rxjs/Observable';
```
Comment (TheDonDope) [inline src/client/app/shared/name-list/name-list.service.spec.ts]: https://angular.io/styleguide#!#coding-conventions
> Import Line Spacing
> Style 03-06
> Do leave one empty line between third party imports and imports of code we created.
> Do list import lines alphabetized by the module.

File: src/client/app/shared/name-list/name-list.service.ts [hunk 6]
```diff
@@ -1,26 +1,51 @@
...
-    'Alan Turing',
-    'Grace Hopper'
-  ];
+  names: string[] = [];
+
+  /**
+   * Contains the currently pending request.
+   * @type {Observable<string[]>}
+   */
+  private _request: Observable<string[]>;
```
Comment (TheDonDope) [inline src/client/app/shared/name-list/name-list.service.ts]: https://angular.io/styleguide#!#coding-conventions
> Properties and Methods
> Style 03-04
> Avoid prefixing private properties and methods with an underscore.

**Comment by TheDonDope** [comment]
Hey @mgechev & @ludohenin

i know i wasn't specifically mentioned to do any review, but i couldn't resist commenting since i just did some StyleGuide relevant refactorisations in the last week and i think it would be nice if we try to keep it consistent to that :) 
But i could also fix those style violations with a separate PR, whichever you think more appropiate :)

Greetings

Dope

File: src/client/app/shared/name-list/name-list.service.spec.ts (hunk 2 above)
Comment (mgechev) [inline src/client/app/shared/name-list/name-list.service.spec.ts]: Yes.

File: src/client/app/app.component.ts (hunk 1 above)
Comment (mgechev) [inline src/client/app/app.component.ts]: Did this following the style guide https://angular.io/styleguide#!#style-07-03.

**Comment by mgechev** [comment]
@TheDonDope thanks for the review! :-)

Let me know if I have missed anything now.

**Comment by TheDonDope** [comment]
@mgechev You are welcome!
I just took a look at the latest changes, looks fine to me now, great work! 👍

**Pull request merged**